### PR TITLE
#145 issues: edited setup.md

### PR DESCRIPTION
**Pull request by quynhngandao**
Edited documentation for setup.md

### CODE REVIEW

**Comment by quynhngandao** [comment]
If this formatting is ok, I will work on the other documentations

**Review by tbenbow — CHANGES_REQUESTED** [review]
Thanks for this PR, it's much needed! All the steps look correct but I think we need some surrounding explanation and many users can probably take the easier "Fork the Repo" route.

File: docs/SETUP.md [hunk 1]
```diff
@@ -1,17 +1,45 @@
 
 # Setup
```
Comment (tbenbow) [inline docs/SETUP.md:2]: Maybe a short intro here.

File: docs/SETUP.md [hunk 2]
```diff
@@ -1,17 +1,45 @@
...
+- Initialize a fresh git repository with the command ```git init```.
+- Add your changes with ```git add .``` and then ```git commit -m 'initial commit’```.
 
-#### 2. Setup Tina Cloud to manage the CMS
-- [Login](https://app.tina.io/signin) to Tina Cloud or [create a new account](https://app.tina.io/register).
-- Go to [projects](https://app.tina.io/projects) and create a new Custom Project.
-- Follow the steps to select your forked repo. (add http://localhost:3000 as the site URL)
-- Click on your new project and save the client ID for later use.
+#### 2. Push Existing Repository to Github 
+- Create a new repo on your GitHub and follow the steps to _“push an existing repository from the command line”_
```
Comment (tbenbow) [inline docs/SETUP.md:11]: I think we can remove steps 1 and 2 and add them to a new "Customizing" page. Only users who are adding custom code that they are not planning to merge into the original Microgen repo need to do things this way. Everyone else can fork the repo.

File: docs/SETUP.md [hunk 3]
```diff
@@ -1,17 +1,45 @@
...
 
-#### 3. Set up a Fleek site
-When you're ready you can setup a site on Fleek running off your repo. You will need to setup environment variables for Tina.
+#### 3. Setting up Tina Cloud to Manage the CMS
+- [Login](https://app.tina.io/signin) to Tina Cloud or [create a new account](https://app.tina.io/register).
+- Go to [projects](https://app.tina.io/projects) and create a new custom project. 
+- Follow the steps to select your forked repo. (make sure to add http://localhost:3000 as the site URL).
+    - Save the client ID of your newly created project for later use.
+
```
Comment (tbenbow) [inline docs/SETUP.md:18]: This section is good but I think we should remove the reference to fork and mention saving the token as well.

File: docs/SETUP.md [hunk 4]
```diff
@@ -1,17 +1,45 @@
...
+#### 4. Setting up a Fleek Site
```
Comment (tbenbow) [inline docs/SETUP.md:19]: I think this section needs a bit of extra explanation. Fleek is one of many possible hosting options (though the only web3 one I know of). We should set this up so others can be added in the future.

File: docs/SETUP.md [hunk 5]
```diff
@@ -1,17 +1,45 @@
...
+- [Login](https://app.fleek.co/#/auth/sign-in?method=Git+Provider) to Fleek or [create a new account](https://app.fleek.co/#/auth/sign-up?method=Git+Provider).
+- Choose **Add New Site** to make a new site and follow the steps to select your forked repo (select _"IPFS"_ as _"Hosting Service"_).
+- Deploy your site by selecting **Deploy Site**.
+    - _Note: The build may be unsuccessful_.
+
+#### 5. Setting up Environment Variables
```
Comment (tbenbow) [inline docs/SETUP.md:25]: We should show an example of the ENV variables here as some users will not be setting the project up locally.

File: docs/SETUP.md [hunk 6]
```diff
@@ -1,17 +1,45 @@
```
Comment (tbenbow) [inline docs/SETUP.md:1]: Thank you for getting all of these steps organized and for going to the trouble of documenting everything!

I added some comments below to your original doc and I copy and pasted your steps into a slightly revised structure here. In general I want this to be the easy version of the setup and we can add some of the trickier bits to a new CUSTOMIZING.md file in another commit.

Can you review the new structured document below and if it looks good use it's structure instead? Feel free to re-word things as you like.

---

# Setup

To get started using Microgen, fork the [repo](https://github.com/pathfindertools/microgen). To deploy your site on the internet, you must connect your repo to the free Tina CMS service and your choice of static webhost.

If you plan to add custom blocks to your site, you may want to consider cloning the repo instead. See the [Customizing](CUSTOMIZING.md) doc for more details.

## Connect to Tina

- [Login](https://app.tina.io/signin) to Tina Cloud or [create a new account](https://app.tina.io/register).
- Go to [projects](https://app.tina.io/projects) and create a new custom project.
- Follow the steps to select your repo (make sure to add <http://localhost:3000> as the site URL).
- Save the Client ID and Token of your newly created project for later use.

## Deploy to a static host

Microgen can be deployed to the static host of your choice. If you want your site to live on Web3, we recommend using [Fleek](https://fleek.co/).

### Fleek

- [Login](https://app.fleek.co/#/auth/sign-in) to Fleek or [create a new account](https://app.fleek.co/#/auth/sign-up).
- Choose **Add New Site** to create a new site and follow the steps to select your forked repo (select _"IPFS"_ as _"Hosting Service"_).
- Go to **Hosting** > **Settings** > **Advanced Build Settings** and add the 2 environment variables from Tina as shown in the Environment Variables section below.
- Go to **Configuration** on Tina Cloud and add the site URL created from Fleek (https://**_<replace-with-your_site-name>_**.on.fleek.co).
- Deploy your site by selecting **Deploy Site**.

### Environment Variables

```
NEXT_PUBLIC_TINA_CLIENT_ID=[YOUR CLIENT ID FROM TINA]
TINA_TOKEN=[YOUR READ ONLY TOKEN FROM TINA]
```

## Making Edits

Once your site has successfully deployed, you can make edits by adding `/admin` to the URL of your site. For example, `https://www.yoursite.com/admin`. Only sites added to the Site URLs section in Tina are allowed to make edits from the admin URL.

## Sync Media

If images are not showing up on your site, you may need to enable and sync media from Tina. Choose media from the sidebar on the Tina site and click "Enable Media", then click the "Sync Media" button.

File: docs/SETUP.md [hunk 7]
```diff
@@ -1,17 +1,45 @@
...
+
+#### 6. Configure Tina Cloud and Fleek
+- Go to **Configuration** on Tina Cloud and add the site URL created from Fleek (https://**_<replace-with-your_site-name>_**.on.fleek.co).
+- Retrigger the deployment on Fleek by going to **Hosting** > **Deploys** and selecting **Trigger Deploy**.
+
+#### 7. Edit Content Through the Website
+- Log in to Tina Cloud through your created website, visit (https://**_<replace-with-your_site-name>_**.on.fleek.co/admin), and follow the steps.
+    - _Note: Any saved changes will be automatically added by Tina_
 
 ## Upgrading MicroGen
```
Comment (tbenbow) [inline docs/SETUP.md:44]: These upgrading instructions assume you have cloned the repo and for many users I think it makes the most sense to fork the repo instead. I think upgrading deserves it's own page and we should have two sets of instructions. One if you've forked and one if you have a clone.

**Review by tbenbow — CHANGES_REQUESTED** [review]
Thanks for this PR, it's much needed! All the steps look correct but I think we need some surrounding explanation and many users can probably take the easier "Fork the Repo" route.

File: docs/SETUP.md (hunk 6 above)
Comment (quynhngandao) [reply docs/SETUP.md]: I've only made minor updates: instead of pointing to a new Customizing.md, it will point to Development.md

**Pull request merged**